RevenueCatUI/Views/PurchaseButton.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  PurchaseButton.swift
+//  
+//
+//  Created by Nacho Soto on 7/18/23.
+//
+
+import RevenueCat
+import SwiftUI
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+struct PurchaseButton: View {
+
+    let package: Package
+    let purchaseHandler: PurchaseHandler
+    let colors: PaywallData.Configuration.Colors
+    let localization: ProcessedLocalizedConfiguration
+    let introEligibility: IntroEligibilityStatus?
+    let mode: PaywallViewMode
+
+    @Environment(\.dismiss)
+    private var dismiss
+
+    var body: some View {
+        self.button
+    }
+
+    private var button: some View {
+        AsyncButton {
+            let cancelled = try await self.purchaseHandler.purchase(package: self.package).userCancelled
+
+            if !cancelled, case .fullScreen = self.mode {
+                self.dismiss()
+            }
+        } label: {
+            IntroEligibilityStateView(
```

Comment:
Feels like this could be more appropriately named something like CTAText view

Reply:
Yeah it's a bad name, but it's not just for the CTA. 
From the docstring:
> /// A view that can process intro eligibility and display different data based on the result.

It's used for CTA but also for other labels. It's just a label that uses a loading intro eligibility and then decides what to display. Better name suggestions welcome.

Reply:
Actually that's what you approved: #2837
I'll hold on in case you have a suggestion for a better name though.

Reply:
Merging this, but let me know if you can think of a better name.